File: lingting-ntp/src/test/java/live/lingting/framework/ntp/NtpTest.java
package live.lingting.framework.ntp;

import org.junit.jupiter.api.Test;

import java.time.LocalDateTime;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertNotNull;
import static org.junit.jupiter.api.Assertions.assertTrue;

/**
 * @author lingting 2024-01-29 17:38
 */
class NtpTest {

	@Test
	void test() throws InterruptedException {
		NtpFactory instance = NtpFactory.INSTANCE;
		Ntp ntp = instance.create();
		assertNotNull(ntp);
		LocalDateTime now = ntp.now();
		assertNotNull(now);
		assertEquals(Ntp.DEFAULT_ZONE_ID, ntp.getZoneId());
		long millis = NtpCn.currentMillis();
		assertTrue(millis > 0);
	}

}

File: lingting-ntp/src/test/kotlin/live/lingting/framework/ntp/NtpTest.kt
package live.lingting.framework.ntp

import live.lingting.framework.time.DatePattern
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Assertions.assertNotNull
import org.junit.jupiter.api.Assertions.assertTrue
import org.junit.jupiter.api.Test

/**
 * @author lingting 2024-01-29 17:38
 */
class NtpTest {
    @Test

    fun test() {
        val ntp = NtpFactory.create()
        assertNotNull(ntp)
        val now = ntp!!.current()
        assertNotNull(now)
        assertEquals(DatePattern.SYSTEM_ZONE_ID, ntp.zoneId)
        val millis = ntp.millis()
        assertTrue(millis > 0)
        assertTrue { ntp.host.isNotBlank() }
        assertNotNull(ntp.diff)
    }
}
